src/main/java/org/ice4j/ice/Component.java

```diff
@@ -1079,11 +1087,20 @@ else if (keepAliveStrategy == KeepAliveStrategy.SELECTED_AND_TCP)
 
     /**
      * Sets the {@link KeepAliveStrategy} for this {@link Component}.
-     * @param strategy the strategy to set.
+     * @param strategy the strategy to set. The keep-alive strategy can only
+     * be modified before the parent {@link Agent} is started.
+     * @throws IllegalStateException if the agent has been started.
      */
     public void setKeepAliveStrategy(KeepAliveStrategy strategy)
     {
+        IceProcessingState agentState
```

Comment:
oops...just saw your other comment as to why @bgrozev.  i still think doing it in the constructor conveys the intention better. if you added a constructor which didn't take the parameter but defaulted it to the current default, would that prevent having to change the other places?